src/main/java/friday/storage/Storage.java

```diff
@@ -0,0 +1,95 @@
+package friday.storage;
+
+import friday.task.Task;
+import friday.task.TaskList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * The Storage class is responsible for handling the reading and writing of task data to a file in the Friday application.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path to the file where task data is stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Checks if the data file exists; if not, creates a new file.
+     *
+     * @throws IOException If an I/O error occurs during file creation.
+     */
+    public void checkFile() throws IOException {
+        File f = new File(filePath);
+        if (!f.exists()) {
+            if (f.createNewFile()) {
+                System.out.println("Data file created.");
+            } else {
+                System.out.println("Error creating data file.");
+            }
+        } else {
+            System.out.println("Loading data from Duke.txt...");
+        }
+    }
+
+    /**
+     * Writes the task list data to the file.
+     *
+     * @param tasks The TaskList containing tasks to be written to the file.
+     * @throws IOException If an I/O error occurs during file writing.
+     */
+    public void writeToFile(TaskList tasks) throws IOException {
+        FileWriter fw = new FileWriter(this.filePath);
+        for (int i = 0; i < tasks.length(); i++) {
+            fw.write(tasks.getTask(i).toString() + System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    /**
+     * Reads task data from the file and returns a TaskList.
+     *
+     * @return The TaskList containing tasks loaded from the file.
+     * @throws IOException If an I/O error occurs during file reading.
+     */
+    public TaskList loadDataFromFile() throws IOException {
+        File file = new File(filePath);
+        TaskList tasks = new TaskList();
+
+        try (Scanner fileScanner = new Scanner(file)) {
+            while (fileScanner.hasNextLine()) {
+                String taskData = fileScanner.nextLine();
+                if (!taskData.isEmpty()) {
```

Comment:
Could have guard clauses instead, to prevent nesting too much?

Reply:
Thx! Will improve!